packages/devtools_app/lib/src/memory/flutter/memory_chart.dart

```diff
@@ -48,44 +42,52 @@ class MemoryChartState extends State<MemoryChart> {
 
   LineChartController get chartController => _chartController;
 
+  final legendTypeFace =
+      TypeFace(fontFamily: 'OpenSans', fontWeight: FontWeight.w100);
+
   @override
   void initState() {
     _initController();
 
-    _setupChart();
-
-    // Read canned data and startup the pseudo-live feed.
-    _initLineData(true);
+    _preloadResources();
 
-    // TODO(terry): Remove when live feed is hooked up.
-    // Hookup to replay the canned data.
-    widget.memoryController.addResetFeedListener(() {
-      setState(() {
-        // Reset our starting index into the canned data and
-        // start feeding the live chart.
-        timerDataIndex = 0;
-      });
-    });
+    _setupChart();
 
     super.initState();
   }
 
   dart_ui.Image _img;
 
-  void _setupChart() async {
+  void _preloadResources() async {
     _img ??= await ImageLoader.loadImage('assets/img/star.png');
   }
 
   @override
   Widget build(BuildContext context) {
+    final MemoryController memoryController = Controllers.of(context).memory;
+
+    if (!memoryController.anyResetFeedListeners) {
+      // Only hookup listeners once.
+      memoryController.addResetFeedListener(() {
+        setState(() {
+          // TODO(terry): TBD Reset allocation stats.
+        });
+      });
+    }
+
+    if (memoryController.memoryTimeline.data.isNotEmpty) {
+      processLiveData();
```

Comment:
This should be handled in a callback listening on the timeline's sampleAdded listener instead of during build.

Reply:
Right its in MemoryChart's listener.